libs/muddle/include/muddle/peer_selection_mode.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+//------------------------------------------------------------------------------
+//
+//   Copyright 2018-2019 Fetch.AI Limited
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//------------------------------------------------------------------------------
+
+namespace fetch {
+namespace muddle {
+
+enum class PeerSelectionMode
+{
+  DEFAULT,  ///< User request based routing
+  KADEMLIA  ///< Kademlia announcing and routing
```

Comment:
just curious, what happens when half of the network are using Kademlia and the other half aren't?

Reply:
So in general we don’t want that to be the case. However, if that does end up being the case, then the system should still function at the same level performance as the previous system.

I.e. performance will largely will depend on the connections that the user requests